webauthn-rs-core/src/core.rs

```diff
@@ -910,8 +964,37 @@ impl WebauthnCore {
             policy
         };
 
-        // Defaults to false.
-        let allow_backup_eligible_upgrade = allow_backup_eligible_upgrade.unwrap_or_default();
+        Ok(ChallengeAuthenticateBuilder {
+            creds,
+            policy,
+            extensions: Default::default(),
+            allow_backup_eligible_upgrade: Default::default(),
+            hints: Default::default(),
+        })
+    }
+
+    /// Generate a new challenge for client authentication from the parameters defined by the
+    /// [ChallengeAuthenticateBuilder].
+    ///
+    /// This function will return the
+    /// [RequestChallengeResponse] which is suitable for serde json serialisation
+    /// to be sent to the client. The client (generally a web browser) will pass this JSON
+    /// structure to the `navigator.credentials.create()` javascript function for registration.
```

Comment:
I'd reword the return part:

---

This function will return:

* a [RequestChallengeResponse], which is sent to the client (and can be serialised as JSON).
 
  A web application would then pass the structure to [the browser's `navigator.credentials.create()` API](https://w3c.github.io/webappsec-credential-management/#dom-credentialscontainer-get) to trigger authentication.

* an [AuthenticationState], which *must* be persisted on the server side. Your application should associate the state with a private session ID, to prevent use in other sessions.